### PR TITLE
Do not remove firewall packages but disable it

### PR DESCRIPTION
**Pull request by rjmateus**
## What does this PR change?

Instead of removing firewall packages which will be needed
by susemanager we now disable the firewall with a parameter
if it is needed.

PR relates to: https://github.com/uyuni-project/sumaform/pull/725

Solves: https://github.com/SUSE/spacewalk/issues/12012

## TODO
- [ ] Run CI with this branch and check results

### CODE REVIEW

**Comment by rjmateus** [comment]
I have run testsuite one time with: proxy, traditional client, minion and ssh minion.
Result:
```
Failing Scenarios:
cucumber features/secondary/trad_migrate_to_minion.feature:95 # Scenario: Cleanup: register minion again as traditional client
cucumber features/secondary/trad_migrate_to_minion.feature:102 # Scenario: Cleanup: check that this is again a traditional client
cucumber features/secondary/trad_migrate_to_minion.feature:107 # Scenario: Cleanup: check that we have again the old activation key
cucumber features/secondary/trad_migrate_to_sshminion.feature:9 # Scenario: Install a package before migration to Salt SSH minion
cucumber features/secondary/trad_migrate_to_sshminion.feature:77 # Scenario: Check that events history is still the same after migration to Salt SSH
cucumber features/secondary/trad_migrate_to_sshminion.feature:106 # Scenario: Cleanup: remove packages from the migrated SSH minion
cucumber features/secondary/srv_delete_channel_with_tool.feature:33 # Scenario: Delete channel with one clone
cucumber features/secondary/srv_delete_channel_with_tool.feature:41 # Scenario: Delete base channel and clone

419 scenarios (8 failed, 113 skipped, 298 passed)
2968 steps (8 failed, 836 skipped, 2124 passed)
```

**Comment by rjmateus** [comment]
cc @jgleissner

File: salt/default/firewall.sls [hunk 1]
```diff
@@ -24,6 +22,7 @@ disable_firewall:
 
 {% else %}
 
+{% if 'server' in grains.get('roles') %}
```
Comment (Bischoff) [inline salt/default/firewall.sls:25]: only for server ? I'm missing why, here...

File: modules/client/variables.tf [hunk 2]
```diff
@@ -31,6 +31,11 @@ variable "use_os_unreleased_updates" {
   default     = false
 }
 
+variable "disable_firewall" {
+  description = "whether to disable the built-in firewall, opening up all ports"
```
Comment (Bischoff) [inline modules/client/variables.tf:35]: could it be you forgot to document this new setting in some README ?

File: salt/default/firewall.sls (hunk 1 above)
Comment (rjmateus) [reply salt/default/firewall.sls:25]: The previous behaviour when running testsuite is configure correctly the firewall on server, and disable it on all other roles. I try to keep it as it was, and only configure the firewall for server

File: modules/client/variables.tf (hunk 2 above)
Comment (rjmateus) [reply modules/client/variables.tf:35]: Thanks. I just add it to readme advanced

File: salt/default/firewall.sls (hunk 1 above)
Comment (Bischoff) [reply salt/default/firewall.sls:25]: ok, thanks for the explanation.

We also configure a firewall on the proxy, but this time from suse manager itself, using a retail formula.

File: modules/client/variables.tf (hunk 2 above)
Comment (Bischoff) [reply modules/client/variables.tf:35]: Thanks.

File: modules/host/main.tf [hunk 3]
```diff
@@ -24,6 +23,9 @@ module "host" {
   provider_settings             = var.provider_settings
   additional_disk_size          = var.additional_disk_size
   volume_provider_settings      = var.volume_provider_settings
+
+  grains = merge({ disable_firewall = var.disable_firewall },
```
[the remark this answers is not in the document]
Comment (rjmateus) [reply modules/host/main.tf:27]: My plan is to move all other variables which are independent from backend to here.
I add this in this way to test the structure/approach, and after it get merged, move all other var. On the TODO list :+1:

[the remark this answers is not in the document]
Comment (moio) [reply modules/host/main.tf:27]: Hmm, that would count as a "disagree and support" from my part ;-)

Go ahead!

**Comment by rjmateus** [comment]
I have tested the deploy locally and it worked.

**Pull request merged**